### ISSUE TITLE
[Windows 11 22000.65] Default Terminal does not work with 1.9!

### ISSUE THREAD

**Issue by obrobrio2000** [microsoft/terminal#10594]
# MAINTAINER NOTE

There's a version mismatch between Terminal 1.9 and 22000.65. We had to introduce a new "default terminal" code path because the one we had introduced for 22000.51 was broken.

Until we release Terminal 1.10 to Preview, this will not work.

---

### Windows Terminal version (or Windows build number)

10.0.22000.65, 1.9.1523.0

### Steps to reproduce

1) Update to Windows 11 22000.65
2) Set Terminal Preview as default terminal app (in Settings > Privacy & security > For developers or in Terminal Preview's settings)
3) Open non-elevated cmd/powershell/wsl.

### Expected Behavior

Terminal Preview should be used as the default terminal, respecting user's choice.

### Actual Behavior

After 22000.65 update, Console Host is always used instead of Terminal Preview (ignoring user's choice). Even if I set Terminal Preview as the default terminal instead of conhost, it always opens everything (cmd, powershell, wsl2 etc.) in conhost.
Feedback Hub link: https://aka.ms/AAd5hc6

**Comment by DHowett** [comment]
Thanks for the report! There's a version mismatch between Terminal 1.9 and 22000.65. We had to introduce a new "default terminal" code path because the one we had introduced for 22000.51 was broken.

Until we release Terminal 1.10 to Preview, this will not work.

Sorry!

**Comment by vaualbus** [comment]
@DHowett  so basically it was done to fix the problem, I reported, that if you set terminal as conhost and you add auto start scripts  in the Startup folders, sometime when the system start you got a windows cannot start app? 
It's a fix anyway at least now i can use again all of my subst that I set at startup, that was broken in prev build!

**Comment by abitofmaya** [comment]
This is also the case on Microsoft Windows [Version 10.0.21390.2025].  Set Terminal Preview as default terminal application but Console Host is used when running Command Prompt or Powershell.

**Comment by obrobrio2000** [comment]
I checked for updates on the Store and it updated Terminal Preview to 1.10.1933.0 , the problem is finally solved! Now cmd, ps, wsl etc. open with Terminal Preview if set as default! Thanks to all the devs!

**Comment by abitofmaya** [comment]
@obrobrio2000, This still does not work on Microsoft Windows [Version 10.0.21390.2025] with Terminal Preview 1.10.1933.0.

**Comment by obrobrio2000** [comment]
> @obrobrio2000, This still does not work on Microsoft Windows [Version 10.0.21390.2025] with Terminal Preview 1.10.1933.0.

Thanks for letting me know, I reopened the issue.

**Comment by DHowett** [comment]
That release predates the revised handoff interface, and I believe that it is an insider build that is not currently receiving updates to any of our components. Our recent work is available in Insider builds 22000 and beyond.

**Comment by ayamadori** [comment]
This still does not work on Microsoft Windows [Version 10.0.22000.71] with Terminal Preview 1.10.1933.0...
![image](https://user-images.githubusercontent.com/7036267/126054736-3995f4ff-aec1-4b59-a88b-3d074bfff55c.png)

**Comment by obrobrio2000** [comment]
> This still does not work on Microsoft Windows [Version 10.0.22000.71] with Terminal Preview 1.10.1933.0...
> ![image](https://user-images.githubusercontent.com/7036267/126054736-3995f4ff-aec1-4b59-a88b-3d074bfff55c.png)
Same as you (22000.71 and 1.10.1993.0) but it seems working fine on my end.

https://user-images.githubusercontent.com/62136803/126055192-b8a85b06-9201-436a-aadb-00348822df47.mp4

**Comment by abitofmaya** [comment]
@obrobrio2000, @ayamadori, Strange. It still does not work on my machine. Are there any steps required other than changing the setting on the Terminal?

**Comment by minpeter** [comment]
windows 11 PRO build 22000.100
windows terminal build 1.10.1933.0
I also have terminal preferences not working. 😥

**Comment by minpeter** [comment]
Rebooting solved it..🤦‍♂️

**Comment by ayamadori** [comment]
This seems to be fixed... (Windows 10.0.22000.120, Terminal Preview 1.10.1933.0)
But I don't know when fixed. :(
When Windows was updated to 22000.120, this should be NOT fixed.
Maybe it's when Visual Studio was installed.

**Comment by The-MAZZTer** [comment]
Not working with Windows 11 22000.160 with Windows Terminal Preview 1.101.933.0

Only unusual thing about my install is I don't have Windows activated (trying it out in a VM). I don't see any messages in Windows Settings indicating that would stop this feature from working though.

I also had installed PowerShell 7, as well as Windows Terminal Stable and Preview, all through winget (shouldn't matter).

Things I tried:
1. Changing default host in Windows Terminal Settings.
2. Verifying default host in Windows Settings, then switching it back and forth to make sure.
3. Rebooting.
4. Running cmd.exe from Run dialog (used console host).
5. Running find.exe/dism.exe from Run dialog (hard to tell but looked like console host window).
6. Created BAT and CMD files with "pause" command. Ran using "start" from a Windows Terminal Command Prompt. (console host).
7. Ran "start cmd" from a Windows Terminal Command Prompt (console host).

**Comment by n-gao** [comment]
With Terminal 1.10 I still can't set it as the default terminal, only the preview works
![image](https://user-images.githubusercontent.com/9084670/131631614-60507f2a-7103-4b7e-9b3e-f224fe8873ca.png)

**Comment by swift-lee** [comment]
> With Terminal 1.10 I still can't set it as the default terminal, only the preview works
> ![image](https://user-images.githubusercontent.com/9084670/131631614-60507f2a-7103-4b7e-9b3e-f224fe8873ca.png)

me too

**Comment by zadjii-msft** [comment]
@n-gao, @swift-lee That's correct. From the [blog post for 1.11](https://devblogs.microsoft.com/commandline/windows-terminal-preview-1-11-release/):

> All of the features from our previous release are now in Windows Terminal, **except for the default terminal setting**, ...

**Comment by zadjii-msft** [comment]
@The-MAZZTer If you're still seeing this, can you share what's in `HKEY_CURRENT_USER\Console\%%Startup` for you? It should look something like 
![image](https://user-images.githubusercontent.com/18356694/131694162-d15a37f1-b185-4d87-bdf2-3f684d515af3.png)

**Comment by The-MAZZTer** [comment]
@zadjii-msft I upgraded to Windows 11 22000.176 and Terminal 1.11.2421.0 and tested again, didn't work.

Checked the registry and I am seeing those values there.

I also verified changing the setting in Windows Settings is actually changing those values.

I checked Event Viewer and found this error, not sure if it is related. It does not correspond with my attempts to open a console host by running find.exe from Run.

Faulting application name: OpenConsole.exe, version: 1.11.2108.30001, time stamp: 0x612d15b6
Faulting module name: uiautomationcore.dll, version: 7.2.22000.1, time stamp: 0x96f1b0d3
Exception code: 0xc0000005
Fault offset: 0x0000000000098096
Faulting process id: 0x1478
Faulting application start time: 0x01d7a03048bf79a6
Faulting application path: C:\Program Files\WindowsApps\Microsoft.WindowsTerminalPreview_1.11.2421.0_x64__8wekyb3d8bbwe\OpenConsole.exe
Faulting module path: C:\Windows\SYSTEM32\uiautomationcore.dll
Report Id: a30f9f02-483a-4e94-b221-156b0eeac623
Faulting package full name: Microsoft.WindowsTerminalPreview_1.11.2421.0_x64__8wekyb3d8bbwe
Faulting package-relative application ID: App

**Comment by zadjii-msft** [comment]
> Faulting module name: uiautomationcore.dll, version: 7.2.22000.1, time stamp: 0x96f1b0d3


HUH. That's a new one. Since the registry does seem to have the right values in it, could you try filing feedback using the feedback hub, and include a trace of trying to run `cmd.exe` from the run dialog? If something's erroring out, that should catch it. 

/feedback

/cc @miniksa

**Comment by Sanluli36li** [comment]
I cannot set the Terminal as the default console application. Is there a way to fix it?

![屏幕截图 2021-09-08 185928](https://user-images.githubusercontent.com/5201743/132498019-4dfdf85f-21a9-41fe-bb67-08b7a390822b.png)
![屏幕截图 2021-09-08 190028](https://user-images.githubusercontent.com/5201743/132498025-8aebc21d-c245-4d59-9231-805541ddc65f.png)

**Comment by zadjii-msft** [comment]
@Sanluli36li You'll need to use the Preview version of the Windows Terminal in order to be able to set it as the default Terminal application. See #11120, https://github.com/microsoft/terminal/issues/10594#issuecomment-910364037

**Comment by DHowett** [comment]
@Sanluli36li your issue probably comes from using the Scoop (or unzipped) version of Terminal. Is that correct?

**Comment by The-MAZZTer** [comment]
@zadjii-msft I should mention I am seeing this on a VM in Oracle VirtualBox. So not sure if my particular instance is something you'd care to pursue or not. Sorry for not mentioning it sooner.

From a fresh boot I ran cmd.exe from the Run dialog and the error I pasted in earlier does not appear in Event Viewer anywhere in the VM session time frame. So it's caused by some other action, presumably something I did while testing different ways to open consoles and switching the default console host setting back and forth. But it doesn't seem related to actually running a console app with Windows Terminal Preview selected as the default,.

One thing I am noticing is when I have Windows Terminal Preview selected, OpenConsole.exe does run when I launch cmd.exe from Run, but it still shows in a conhost window. OpenConsole.exe for some reason hangs around a minute or two after I close the conhost window (in Windows 10 it exits immediately once a Windows Terminal-hosted console is closed).

If I switch back to conhost as the default host, OpenConsole.exe does not run for cmd.exe from the Run dialog (just like on Windows 10).

When using Windows Terminal Preview OpenConsole reacts the same way it does on Windows 10... it exits immediately once a console closes.

I've filed Feedback with the link to this issue and attached a recording of me attempting to run cmd.exe from Run in case there is something useful in there. Oddly Feedback did not allow me to select "Windows Terminal (Preview)" (no option for it) so I just selected "Windows Terminal".

**Comment by zadjii-msft** [comment]
> One thing I am noticing is when I have Windows Terminal Preview selected, OpenConsole.exe does run when I launch cmd.exe from Run, but it still shows in a conhost window. OpenConsole.exe for some reason hangs around a minute or two after I close the conhost window (in Windows 10 it exits immediately once a Windows Terminal-hosted console is closed).

VERY interesting. That seems to suggest that the delegation _console_ part of defterm is working as expected, but the delegation _terminal_ aspect isn't. The reason you're seeing what appears to be a conhost window is because OpenConsole.exe _is literally just conhost.exe_. It's just the version that's built from this repo, so we can ship an updated version side-by-side with the Terminal.

@miniksa for insights, he's the defterm expert.

**Comment by miniksa** [comment]
@The-MAZZTer... what version of VirtualBox? The best thing would be if I could get a reproduction of this issue in my office.

I'm going to search for the feedback but I don't know if I'll be able to find it without the share link. Wish me luck.

**Comment by miniksa** [comment]
Oh my god I found it. Lightspeed ID 32f6166c-7884-4494-b2f0-8a29523bb95d

**Comment by miniksa** [comment]
OK there's no Terminal crash data associated with this, so I'm not sure why the delegation _terminal_ didn't work unless it was just not found or something. There wasn't an exit code that said *why*. Short of adding a ton of logging infrastructure to this to try to suss it out... the best thing I could do is get a reproduction of the environment somewhere I can have a debugger on it.

**Comment by drrrMikado** [comment]
> @n-gao, @swift-lee That's correct. From the [blog post for 1.11](https://devblogs.microsoft.com/commandline/windows-terminal-preview-1-11-release/):
> 
> > All of the features from our previous release are now in Windows Terminal, **except for the default terminal setting**, ...

Is there a plan to set the default terminal in the Stable version?

**Comment by DHowett** [comment]
> Is there a plan to set the default terminal in the Stable version?

Yes, when the feature is stable.

**Comment by jerryc127** [comment]
The default Terminal only for windows 11 ?

In windows 10, i have not this option in 1.11.2421.0

![image](https://user-images.githubusercontent.com/16351105/133921619-02ab1dc9-1f6d-4315-96e6-5d5050c876e1.png)

**Comment by minpeter** [comment]
> The default Terminal only for windows 11 ?
> 
> In windows 10, i have not this option in 1.11.2421.0
> 
> ![image](https://user-images.githubusercontent.com/16351105/133921619-02ab1dc9-1f6d-4315-96e6-5d5050c876e1.png)
> 
> 
Is the 'windows  terminal preview' correct?

**Comment by jerryc127** [comment]
> > The default Terminal only for windows 11 ?
> > In windows 10, i have not this option in 1.11.2421.0
> > ![image](https://user-images.githubusercontent.com/16351105/133921619-02ab1dc9-1f6d-4315-96e6-5d5050c876e1.png)
> 
> Is the 'windows terminal preview' correct?

yes
![image](https://user-images.githubusercontent.com/16351105/133921937-4cb459e7-056a-4494-a017-7baf18a430dd.png)

**Comment by zadjii-msft** [comment]
> The default Terminal only for windows 11 ?
> 
> In windows 10, i have not this option in 1.11.2421.0

That's correct, this feature is only available in Windows 11, when you're using the Preview version of the Terminal.

**Comment by The-MAZZTer** [comment]
I have reinstalled Windows 11 in a VM from (mostly) scratch to create a reproduction case.

One thing I really should have mentioned is this involves third-party tools; I THINK they just use MS tools internally but I'm not sure. It's entirely possible these tools are responsible for this problem (but I don't see how; I would expect something more catastrophic if the tool produced something broken).

1. I am using VirtualBox Version 6.1.26 r145957 (Qt5.6.2) under Windows 10 21H1. I have Hyper-V Hypervisor disabled to allow VirtualBox to run better.
2. I downloaded a script from uupdump.net to build a Windows 11 ISO. https://uupdump.net/download.php?id=7fbfaec5-bd63-4532-b022-22eb9247a57e&pack=en-us&edition=professional
3. I booted my existing Windows 11 VM into the ISO
4. I used the Repair options from Windows Setup to run regedit and add bypass registry values to ignore TPM/SecureBoot requirements. I added the registry values listed here: https://appuals.com/bypass-windows-11-installation-requirements/
5. I ran setup.exe from cmd to continue Windows Setup. I formatted and installed over the Windows partition from my previous Windows 11 VM.
6. I booted into Windows 11 and completed the OoB experience wizard with a local account.
7. I opened Microsoft Store and installed Windows Terminal Preview.
8. I launched Windows Terminal Preview and opened Settings, and configured it to use Windows Terminal Preview for the default console host.
9. I hit Win+R and ran cmd.exe
10. Command Prompt opened in a console host window instead of the expected Windows Terminal Preview.

Hopefully this will be helpful for reproduction.

**Comment by miniksa** [comment]
FWIW, I tried to install and use VirtualBox 6.1.26 on my Windows 11 host machine and it couldn't start any VMs. I wonder if that's a known issue for VirtualBox and I need a Windows 10 host machine?

**Comment by CDAGaming** [comment]
Any updates on this?

Given the release of Windows 11 is under 24 hours away at time of writing, I'd love to know whether this was nearing stability...(On the fence about using the preview version of the terminal, since the only thing I'm after is the default setting)

**Comment by zadjii-msft** [comment]
@CDAGaming This isn't the thread for that discussion, though I'm not sure there is another one. This one's tracking a very specific bug in the defterm implementation that we're still trying to track down, but there's only the one user with the repro.

I'm gonna file another issue to track that discussion in the immediate future.

**Comment by zadjii-msft** [comment]
If you want to discuss that topic more, let's move over to #11415.

We still haven't found anything that looks like a root cause for this one, nor do we have any live repros unfortunately.

**Comment by shaunthorne** [comment]
> This seems to be fixed... (Windows 10.0.22000.120, Terminal Preview 1.10.1933.0) But I don't know when fixed. 🤔 When Windows was updated to 22000.120, this should be NOT fixed. Maybe it's when Visual Studio was installed.

Windows 11 Build 2200.194. Windows Terminal Preview 1.11.2731.0

Windows Terminal started to work as expected immediately after installing Visual Studio 2019 with .NET components
Is it kind of magic?
![image](https://user-images.githubusercontent.com/52523716/136463059-a5af1956-f693-4e00-9ffb-97850822b641.png)

**Comment by zadjii-msft** [comment]
@The-MAZZTer These questions might seem silly, but I'm just trying to correlate with an internal report of a similar bug. Are you by any chance running on an ARM CPU? 

Even more wildly: Is the copy of Windows you're running in VirtualBox activated?

We've got a crazy theory that the second bit might somehow be causing this to fail (but we've still got NO idea why that might be)

**Comment by NotTheDr01ds** [comment]
@zadjii-msft While we're waiting on The-MAZZTer's reply, I will chime in that I can also reproduce this with:

* An activated Windows 11 (22000.258)
* Installed on KVM/QEMU on a Linux (openSUSE Tumbleweed)
* Completed initial install with no network, so local account (sounds similar to The-MAZZTer's process)
* Did not activate during install
* After reboot, installed virtio drivers
* Connected to network
* Activated Windows
* Converted to Microsoft Account
* Installed WSL with Ubuntu, Debian, and Alpine
* Installed Windows Terminal Preview (currently 1.11.2731.0)
* Set it to default through the Windows "For Developers" settings
* Ran `cmd.exe`
* Opened in Windows Console Host
* Rebooted
* Ran `cmd.exe`
* Opened in Windows Console Host

**Comment by The-MAZZTer** [comment]
@zadjii-msft Missed your other questions, I'm running on a x64 host (is there even an ARM VirtualBox?) and I didn't activate Windows 11 since I don't have a license, apart from a couple Windows 10 ones (just wanted to take a sneak peek before I update my Windows 10 PC).

**Comment by zadjii-msft** [comment]
@The-MAZZTer Oh sorry, we moved on past those theories. We believe this was due to an accidental dependency on the CRT. Manually installing the [C Runtime](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-160) should fix this (until we roll out a hotfix build of the Terminal that doesn't have this dependency anymore)
